File: backend/app/schemas/device.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class DeviceCreate(BaseModel):
    device_code: str = Field(..., max_length=50)
    name: str = Field(..., max_length=100)
    location: Optional[str] = Field(None, max_length=200)
    description: Optional[str] = None


class DeviceUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=100)
    location: Optional[str] = Field(None, max_length=200)
    status: Optional[int] = Field(None, ge=0, le=2)
    description: Optional[str] = None


class DeviceResponse(BaseModel):
    id: int
    device_code: str
    name: str
    location: Optional[str]
    status: int
    last_heartbeat: Optional[datetime]
    description: Optional[str]
    created_at: Optional[datetime]
    updated_at: Optional[datetime]

    class Config:
        from_attributes = True

File: frontend/src/types/device.ts
export interface Device {
  id: number
  device_code: string
  name: string
  location?: string
  description?: string
  status: number // 0=offline, 1=online, 2=maintenance
  is_online?: boolean
  last_heartbeat?: string
  created_at?: string
  updated_at?: string
}

export interface DeviceListResponse {
  items: Device[]
  total: number
  page: number
  page_size: number
}

export interface DeviceCreate {
  device_code: string
  name: string
  location?: string
  description?: string
}

export interface DeviceUpdate {
  name?: string
  location?: string
  status?: number
  description?: string
}